cmd/nodeagent/nodeagentmanager.go

```diff
@@ -530,12 +538,17 @@ func (n *NodeAgentManager) createGameServerDetails(ctx context.Context, gsuid ty
 		return err
 	}
 
+	spec, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&gsd.Spec)
```

Comment:
Thanks for catching that.
I'd assume this works right now since the GameServerDetail is created when the allocation takes place, when ConnectedPlayers is empty. However, in case this fails and the GameServerDetail creation will take place in a subsequent heartbeat, we'd need to include the existing ConnectedPlayers from the heartbeat.
Trying to think of a good way to test this, but it looks hard.

Reply:
However, now that I think of it, this flow will happen with the existing code:
- allocation happens
- GameServerDetail creation fails
- subsequent heartbeat succeeds but doesn't include the .Spec (ConnectedPlayers)
- user will eventually send new ConnectedPlayers so the update will take place

taking the above into account, don't think we need a hotfix for now. Thoughts?

Reply:
Yeah, for a while I added a value I needed to the spec of the details, but I moved it to labels, so this stayed there.

Reply:
Labels is good, we can add it to .Spec later if needed. Thanks!